File: update_galleries.py
import re
import os

with open('src/data/stranda-apartments.ts', 'r') as f:
    content = f.read()

keys = re.findall(r"'([A-Za-z0-9_-]+)':\s*\{", content)

public_dir = 'public/images/stranda'

for key in keys:
    key_idx = content.find(f"'{key}': {{")
    if key_idx == -1: continue
    
    next_key_match = re.search(r"\n\s*'([A-Za-z0-9_-]+)':\s*\{", content[key_idx + 10:])
    next_key_idx = len(content) if not next_key_match else key_idx + 10 + next_key_match.start()
    
    block = content[key_idx:next_key_idx]
    
    gallery_match = re.search(r"(gallery:\s*\{.*?\"?images\"?:\s*\[.*?\]\s*\})", block, re.DOTALL)
    if not gallery_match:
        print("No gallery found for", key)
        continue
    
    dirs_to_try = [key, key + '_images', key.lower(), key.replace('_', '-'), key.replace('-', '_'), 'C_Generic', 'Pokoje']
    if key == 'c-studio' or key == 'C_Studio': dirs_to_try.insert(0, 'C_Studio')
    if key == 'Studio': dirs_to_try.insert(0, 'C_Studio')
    
    target_dir = None
    for d in dirs_to_try:
        p = os.path.join(public_dir, d)
        if os.path.isdir(p):
            target_dir = p
            break
            
    if not target_dir:
        print("No dir for", key)
        continue
        
    webp_files = []
    for root, _, files in os.walk(target_dir):
        for file in files:
            if file.endswith('.webp'):
                full_path = os.path.join(root, file)
                rel_path = '/' + os.path.relpath(full_path, 'public')
                webp_files.append(rel_path)
                
    if not webp_files:
        print("No webp files for", key)
        continue
        
    def sort_key(f):
        f = f.lower()
        if 'hero' in f: return (0, f)
        
        import re as regex
        num = regex.search(r'_(\d+)\.webp$', f)
        if num: return (2, int(num.group(1)), f)
        
        if '_1.' in f: return (1, 0, f)
        return (3, 0, f)
        
    webp_files.sort(key=sort_key)
    
    hero = webp_files[0]
    images_str = ',\n'.join([f'                getAssetPath("{f}")' for f in webp_files])
    
    replacement = f'''gallery: {{
            "heroImage": getAssetPath("{hero}"),
            "images": [
{images_str}
            ]
        }}'''
        
    new_block = block.replace(gallery_match.group(1), replacement, 1)
    content = content[:key_idx] + new_block + content[next_key_idx:]
    print(f"Updated {key} with {len(webp_files)} images.")

with open('src/data/stranda-apartments.ts', 'w') as f:
    f.write(content)

File: update_galleries.js
const fs = require('fs');
const path = require('path');

const tsFilePath = path.join(__dirname, 'src', 'data', 'stranda-apartments.ts');
let tsContent = fs.readFileSync(tsFilePath, 'utf8');

const publicDir = path.join(__dirname, 'public', 'images', 'stranda');

function getAllWebp(dir, relativeTo) {
    let results = [];
    if (!fs.existsSync(dir)) return results;
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);
        if (stat && stat.isDirectory()) {
            results = results.concat(getAllWebp(fullPath, relativeTo));
        } else if (fullPath.endsWith('.webp')) {
            results.push('/' + path.relative(relativeTo, fullPath));
        }
    });
    return results;
}

const keyRegex = /'([A-Za-z0-9_-]+)':\s*\{/g;
let match;
const keys = [];
while ((match = keyRegex.exec(tsContent)) !== null) {
    keys.push(match[1]);
}

let modified = tsContent;

for (const key of keys) {
    let dirToSearch = null;
    
    // c-studio maps to c_studio or c-studio
    let baseKey = key;
    if (key === 'c-studio') baseKey = 'C_Studio';
    
    const possibleDirs = [
        baseKey,
        key,
        key + '_images',
        key.toLowerCase(),
        key.replace(/_/g, '-'),
        key.replace(/-/g, '_'),
        'C_Generic',
        'Pokoje'
    ];
    
    for (const p of possibleDirs) {
        const candidate = path.join(publicDir, p);
        if (fs.existsSync(candidate) && fs.statSync(candidate).isDirectory()) {
            dirToSearch = candidate;
            break;
        }
    }

    if (dirToSearch) {
        let webpFiles = getAllWebp(dirToSearch, path.join(__dirname, 'public'));
        
        if (webpFiles.length > 0) {
            webpFiles.sort((a, b) => {
                const aName = path.basename(a).toLowerCase();
                const bName = path.basename(b).toLowerCase();
                if (aName.includes('hero') && !bName.includes('hero')) return -1;
                if (!aName.includes('hero') && bName.includes('hero')) return 1;
                
                const numA = parseInt((aName.match(/_(\d+)\.webp$/) || [0, '9999'])[1], 10);
                const numB = parseInt((bName.match(/_(\d+)\.webp$/) || [0, '9999'])[1], 10);
                
                if (numA !== numB) return numA - numB;
                
                if (aName.includes('_1.') && !bName.includes('_1.')) return -1;
                if (!aName.includes('_1.') && bName.includes('_1.')) return 1;
                return a.localeCompare(b);
            });

            const hero = webpFiles[0];
            const imagesStr = webpFiles.map(f => `                getAssetPath("${f}")`).join(',\n');
            
            const galleryReplacement = `gallery: {
            "heroImage": getAssetPath("${hero}"),
            "images": [
${imagesStr}
            ]
        }`;

            // Find key block
            const blockStartIdx = modified.indexOf(`'${key}': {`);
            if (blockStartIdx !== -1) {
                // Find next key block
                const nextBlockIdx = modified.indexOf(`': {`, blockStartIdx + 10);
                let searchRegion = nextBlockIdx !== -1 ? modified.substring(blockStartIdx, nextBlockIdx) : modified.substring(blockStartIdx);
                
                // Manually find gallery: { ... } by bracket counting
                const galleryStart = searchRegion.indexOf('gallery: {');
                if (galleryStart !== -1) {
                    let braces = 0;
                    let galleryEnd = -1;
                    for (let i = galleryStart + 8; i < searchRegion.length; i++) {
                        if (searchRegion[i] === '{') braces++;
                        if (searchRegion[i] === '}') {
                            braces--;
                            if (braces === 0) {
                                galleryEnd = i + 1;
                                break;
                            }
                        }
                    }
                    if (galleryEnd !== -1) {
                        const originalGallery = searchRegion.substring(galleryStart, galleryEnd);
                        modified = modified.replace(originalGallery, galleryReplacement);
                        console.log(`Updated gallery for ${key} with ${webpFiles.length} images.`);
                    }
                }
            }
        }
    }
}

fs.writeFileSync(tsFilePath, modified, 'utf8');
console.log('Update complete.');
